### PR TITLE
Add http backend to origin data export API

### LINKED ISSUE #1180: Include https backend in the origin export API for the UI
We added support for origins to take https storage backend recently #958 , and we want to expose this type of backend to the origin export API and correctly display the related values

@CannonLock , since S3/POSIX/http or the future Globus backend each has a different combination of `Origin.Exports` entries and other configurations to set up (i.e. `Origin.S3Bucket` for s3 backend and `Origin.HttpServiceUrl` for `http` backend), do we want to render different panels based on different storage type?

### PR DESCRIPTION
**Pull request by haoming29**
Closes #1180 

This added extra fields to the `/origin_ui/exports`, additional UI work required to render the data.

### CODE REVIEW

**Comment by haoming29** [comment]
@jhiemstrawisc Let's hold off merging this PR until your #1285 PR gets merged in. I want to include `xroot` backend in this pass as well.

File: swagger/pelican-swagger.yaml
Comment (jhiemstrawisc) [inline swagger/pelican-swagger.yaml:482]: Let's remove references to AWS, since this is a core part of the protocol (as opposed to the service offered by Amazon)

File: swagger/pelican-swagger.yaml
Comment (jhiemstrawisc) [inline swagger/pelican-swagger.yaml:488]: Is it worth providing an example of the difference between the two here, or are you happy with what's in our docs elsewhere?

File: swagger/pelican-swagger.yaml
Comment (jhiemstrawisc) [inline swagger/pelican-swagger.yaml:479]: Not a question we need to answer in this PR, but I wonder what our long-term strategy is for enumerating backends in places like this. Ideally we want to add many more backends, but it seems like we list available options manually in a variety of locations. It might be hard to keep things up to date if we don't come up with a solution.

File: docs/parameters.yaml
Comment (jhiemstrawisc) [inline docs/parameters.yaml:480]: Ahh -- S3Bucket isn't actually required. When we want to export an entire endpoint like amazon's open data registry, we can't manually enumerate all of the buckets. In that case we get around it by omitting the bucket (which then becomes a part of the object name)

**Review by jhiemstrawisc — CHANGES_REQUESTED** [review]

File: swagger/pelican-swagger.yaml
Comment (haoming29) [reply swagger/pelican-swagger.yaml:488]: I think this is mostly a rough intro for people looking up the API for what they'll get out of that endpoint. If they are interested in learning what exactly the value is for, they can refer to the docs website. I'll link to it

File: swagger/pelican-swagger.yaml
Comment (haoming29) [reply swagger/pelican-swagger.yaml:479]: Yeah I had the same question. I don't know how we can sync Swagger docs with `parameters.yaml` and other places, but I'd be nice to document what to change for an enum type configuration

**Review by jhiemstrawisc — APPROVED** [review]
LGTM

**Pull request merged**
